File: sol/solution/0001-0100/0024/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        curr = dummy
        while curr.next and curr.next.next:
            first = curr.next
            second = curr.next.next
            first.next = second.next
            second.next = first
            curr.next = second
            curr = first
        return dummy.next

File: sol/solution/0001-0100/0024/Solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* swapPairs(struct ListNode* head) {
    struct ListNode dummy;
    dummy.next = head;
    struct ListNode* curr = &dummy;
    while (curr->next && curr->next->next) {
        struct ListNode* first = curr->next;
        struct ListNode* second = curr->next->next;
        first->next = second->next;
        second->next = first;
        curr->next = second;
        curr = first;
    }
    return dummy.next;
}